preprocessing/write_chars.py

```diff
@@ -1,14 +1,18 @@
-# Create a string of chars from char code 32 to 127,
-# in 12 rows and 8 columns with a blank last entry.
+"""Create a string of chars from char code 32 to 127,
+in 12 rows and 8 columns with a blank last entry.
 
-# This should be screenshotted (e.g. chars.png)
-# and used with analyze_chars.py.
+This should be screenshotted (e.g. chars.png)
+and used with analyze_chars.py."""
 
-output = ''
+def add_new_lines(input, every):
+	"""Add a new line into the string 'input' at a frequency of 'every'.
+	e.g. add_new_lines("foo", 1) -> 'f\\no\\no'."""
+	return '\n'.join(input[i:i+every] for i in range(0, len(input), every))
 
-for i in range(32, 128):
-    output += chr(i)
-    if i % 8 == 0:
-        output += '\n'
-        
-print(output)
\ No newline at end of file
+def main():
+	"""Print chars which will be used to construct ASCII art."""
+	all_characters = ''.join(map(range(32, 128), chr))
```

Comment:
do let me know if any of the lines are hard to understand, 

ideally, (IMO), rather than hard code being all in the same function, hard code should be extracted to a method with an informative name, and now if theres an error i can choose to inspect `add_new_lines`, but at least i know for sure what it does. This is part of a concept called self-documenting code, so you don't need comments.